code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -615,12 +615,15 @@
 	if(iscarbon(M) && M.stat != DEAD)
 		var/mob/living/carbon/C = M
 		var/holycheck = ishumanbasic(C)
-		if(!(holycheck || islizard(C)) || reac_volume < 5) // implying xenohumans are holy //as with all things,
+		if(!(holycheck || islizard(C) || ismoth(C)) || C.dna.features["moth_wings"] == "Burnt Off" || reac_volume < 5 ) // implying xenohumans are holy //as with all things,
```

Comment:
Why having both `ismoth(C)) || C.dna.features["moth_wings"] == "Burnt Off"` ?
Won't the first check always be true if the second is?

Reply:
The first group of checks is just to make sure they are one of the races that can have potion of flight wings, and then it checks if they have burnt off moth wings or if the reaction volume is too low to continue.
So if they're not a human, lizard, or moth it will fail, and if they have burnt off moth wings, or they take <5u it will fail.

Reply:
Then you meant something like `(ismoth(C)) && C.dna.features["moth_wings"] != "Burnt Off")`
That said, instead of doing several is-checks you could do a redefined proc, something like `if(!C.species.is_affected_by_flightpotion() || reac_volume < 5)`

Reply:
I never meant it to check both if you are a moth and if you have burnt off moth wings, as it's checking if you are not one of the valid races, and then checking if you have burnt off moth wings. 
Obviously if you're not a moth you won't have those wings, and if you are a moth then you'll get past the race checks to have your moth_wings checked. 
`!(holycheck || islizard(C) || ismoth(C))` is solely being used to make sure that no races that don't specifically have flight potion wings made for them will go beyond the early return.

I'd imagine I could manage to get a proc put together that would check these things and check off of that, but I have no idea if that'd be more efficient than checking if they meet any of these fail conditions in this proc. 
If you'd like I can try that. I just wanna be sure that's what you're trying to get across.

Reply:
These 3 are species checks, correct?
What I was suggesting was running a species proc, something like `C.species.is_affected_by_flightpotion()`

If you'd rather not, then at least limit the moth wings check to moths with something like this:
`if(reac_volume < 5 || !(holycheck || islizard(C) || (ismoth(C) && C.dna.features["moth_wings"] != "Burnt Off" )))`